### PR TITLE
feat: adding configuration to provide exclusion for failOnError with http post processor

### PR DESCRIPTION
**Pull request by mayankrai09**
feat: Added exclusion range with fail_on_errors for HTTP post-processors
GTF side we have some use cases where we must ignore some of the HTTP response codes when fail_on_errors is true.

### CODE REVIEW

File: dagger-core/src/main/java/com/gotocompany/dagger/core/processors/external/http/HttpResponseHandler.java [hunk 1]
```diff
@@ -123,19 +130,37 @@ private void successHandler(Response response) {
...
     }
 
+    private boolean shouldFailOnError(Integer statusCode) {
+        if (statusCode == 0 || StringUtil.isNullOrEmpty(httpSourceConfig.getExcludeFailOnErrorsCodeRange())) {
+            return true;
+        }
+        return !getExcludeFailOnErrorCodes(httpSourceConfig.getExcludeFailOnErrorsCodeRange()).contains(statusCode);
+    }
+
+    private HashSet<Integer> getExcludeFailOnErrorCodes(String input) {
```
Comment (lavkesh) [inline dagger-core/src/main/java/com/gotocompany/dagger/core/processors/external/http/HttpResponseHandler.java:154]: We should save the list of error codes and not generate for every response. the list can be created and saved outside of the Response Handler.

File: dagger-core/src/main/java/com/gotocompany/dagger/core/processors/external/http/HttpResponseHandler.java [hunk 2]
```diff
@@ -86,15 +91,17 @@ public Object onCompleted(Response response) {
             successHandler(response);
         } else {
             postResponseTelemetry.validateResponseCode(meterStatsManager, statusCode);
-            failureHandler("Received status code : " + statusCode);
+            boolean shouldFailOnError = (httpSourceConfig.isFailOnErrors() ? shouldFailOnError(statusCode) : false);
+            failureHandler("Received status code : " + statusCode, shouldFailOnError);
```
Comment (lavkesh) [inline dagger-core/src/main/java/com/gotocompany/dagger/core/processors/external/http/HttpResponseHandler.java:95]: i would suggest to keep the method signature same. 
and in line 131 do 

if( shouldFailOnError(statusCode)) {
...
}

then the method sholdFailOnError will do the necessary checks

Comment (mayankrai09) [reply dagger-core/src/main/java/com/gotocompany/dagger/core/processors/external/http/HttpResponseHandler.java]: Made changes

File: dagger-core/src/main/java/com/gotocompany/dagger/core/processors/external/http/HttpResponseHandler.java (hunk 1 above)
Comment (mayankrai09) [reply dagger-core/src/main/java/com/gotocompany/dagger/core/processors/external/http/HttpResponseHandler.java]: agree, made changes

**Review by lavkesh — APPROVED** [review]

**Pull request merged**